Add automated and manual deployment docs

rogeruiz opened the pull request:
~~This PR is a _Work In Progress_~~

This patch adds documentation around manual and automated deployments for 
- visibility into the automated deployment process ( _which branch auto-deploys to which URL_? )
- steps to run a manual deployment ( _if necessary_ ? )

Resolves two items from #33

rogeruiz committed:
Add automated and manual deployment docs

rogeruiz commented on README.md:
````diff
@@ -90,10 +90,72 @@ To start the local webserver and have gulp watch for changes:
 ```
 
-#### Descriptions for optional `gulp [ flags ]` task prefixes
+### Descriptions for optional `gulp [ flags ]` task prefixes
````
This increase in heading along with the one above it for `Development` allow for better typographic headers in the `README.md`.

rogeruiz commented on README.md:
```diff
@@ -90,10 +90,72 @@ To start the local webserver and have gulp watch for changes:
+[cg-deploy-hw]: https://docs.cloud.gov/getting-started/your-first-deploy/ "Cloud.gov: Your First Deploy"
+[cg-deploy-ss]: https://docs.cloud.gov/apps/static/ "Cloud.gov: Deploying Static Sites"
+
+> Manual deployments aren't necessary as all updates should go through Travis-CI.
```
Should this be more prominent? I wasn't sure if the `Manual deployment` section is necessary if we're going to stick to `Automated deployment`. but this information is valuable in case `Travis-CI` is down. Thoughts @juliaelman @maya @yozlet? Let me know if we want to add anymore to this section as well. :bow:

rogeruiz replied:
switch `as all updates` with `as all deployments` ?

juliasolorzano replied:
It's nice to have the manual deployment option documented as a back up deployment option. 

:+1: for `as all deployments`

rogeruiz commented on README.md:
```diff
@@ -90,10 +90,72 @@ To start the local webserver and have gulp watch for changes:
+[ffd-staging]: https://ffd-microsite-staging.apps.cloud.gov "Federal Front Door: Staging"
+[ffd-production]: https://labs.usa.gov "Federal Front Door: Production"
+
+All new Pull Requests must target this repository's `staging` branch. You can
```
> All new Pull Requests must target this repository's `staging` branch. You can

:thought_balloon: This seems more like `CONTRIBUTING` to me than related to `Automated deployments` :thought_balloon:

juliasolorzano replied:
:+1: agreed.

rogeruiz committed:
Update `production` flag documentation

rogeruiz committed:
Update grammar and language around deployments

rogeruiz committed:
Remove docs better suited for `CONTRIBUTING`

rogeruiz committed:
Add table of contents

rogeruiz committed:
Update language around assumptions;

This patch updates the language around assumptions and creates a list in
case we need to make _evermore assumptions_.

rogeruiz commented:
@juliaelman I've updated the documentation with your suggestions, grammar fixes, and added a `Table of Contents`. :thought_balloon:  I think the `[ work in progress ]` prefix can be removed on this one. :thought_balloon:

rogeruiz committed:
Update table formatting for gh-flavored markdown;

Source => https://guides.github.com/features/mastering-markdown/#GitHub-flavored-markdown

yozlet commented:
Looks fab to me! Thank you, Roger! :+1:
